homeassistant/components/foscam/__init__.py

```diff
@@ -21,9 +25,23 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up foscam from a config entry."""
-    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = entry.data
+    session = FoscamCamera(
+        entry.data[CONF_HOST],
+        entry.data[CONF_PORT],
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        verbose=False,
+    )
+    coordinator = FoscamCoordinator(hass, session)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "coordinator": coordinator,
+    }
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```
Don't think we need to store anything else here so can put the coordinator directly